### PR TITLE
FormCommit: Read committer info using git

### PR DESCRIPTION
**Pull request by mstv**
Fixes https://github.com/gitextensions/gitextensions/issues/5492#issuecomment-1448460964

## Proposed changes

- FormCommit: Read committer info using git,
  use `RunAsync` and
  do not cache the git output in order to update on FormActivate (existing testcase)

## Screenshots 

unchanged

## Test methodology 

- manual
- existing tests

## Merge strategy

I agree that the maintainer squash merge this PR (if the commit message is clear).

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

File: GitCommands/Git/GitModule.cs
Comment (gerhardol) [inline GitCommands/Git/GitModule.cs:2316]: Use a separate function instead?

**Review by gerhardol — COMMENTED** [review]
+1
This increases the startup time for the first start of FormCommit with about 150 ms.
Annoying...
The cache should probably be cleared when exiting settings form.

File: GitCommands/Git/GitModule.cs
Comment (RussKie) [reply GitCommands/Git/GitModule.cs:2316]: Yes please. Let's call it `GetEffectiveGitSetting()`. Cache by default until we have a real use case that requires no caching.

**Review by RussKie — COMMENTED** [review]
Tests simulating the reported scenario would be great to have.

**Comment by RussKie** [comment]
Should the setting be reloaded when a repo is switched?

**Comment by vbjay** [comment]
> Should the setting be reloaded when a repo is switched?

Yes. Config in other repo can be different.

**Comment by mstv** [comment]
> This increases the startup time for the first start of FormCommit with about 150 ms.

I am going to do it async.

> The cache should probably be cleared when exiting settings form.

good point

> > Should the setting be reloaded when a repo is switched?
> 
> Yes. Config in other repo can be different.

I thought we do so already - and was wrong again as in a discussion longer ago. I am adding a non-static cache.

> Tests simulating the reported scenario would be great to have.

I am going to adapt the existing tests as far as necessary.
My interest in this feature is very limited though. I just see the necessity to correctly display the info. Let's leave it up to git.
I hope @Gutza will verify it. Could you setup an integration testcase?

**Comment by mstv** [comment]
> I am adding a non-static cache.

All current usages of the `GitModule.GitCommandCache` seem to be repo-specific as far as I can see.
So, having only a static instance is not correct for these.
Are there objections turning it into a member instance (other than the binding in FormGitCommandLog)?

BTW: Is a cache entry really such expensive that we limit the MRU cache to 50 entries?

File: GitUI/CommandsDialogs/FormCommit.cs
Comment (gerhardol) [inline GitUI/CommandsDialogs/FormCommit.cs:2833]: ?
```suggestion
                    commitAuthorStatus.Text = string.IsNullOrWhitespace(toolAuthor.Text))
```

**Review by gerhardol — APPROVED** [review]
Have not run, but it looks fine.

**Comment by mstv** [comment]
> Let's call it `GetEffectiveGitSetting()`.

Thank you for directly proposing a name, Igor.

> Cache by default until we have a real use case that requires no caching.

I agree. Caching should be the default for settings.
And we already have a use case: The committer info shall be updated on FormActivate as per existing testcase.

So, since this PR does not depend on caching (`RunAsync` is the key), this PR would be ready if AppVeyor would be able to run Form tests without headstands. 
Any idea how to make `Should_update_committer_info_on_form_activated` pass?

**Comment by mstv** [comment]
Sorry AppVeyor, I have found where to place the missing wait.

**Comment by mstv** [comment]
Does https://ci.appveyor.com/api/buildjobs/whg8axsj2xdhj86v/artifacts/artifacts%2FRelease%2Fpublish%2FGitExtensions-Portable-4.1.0.16396-f9631e23c.zip work for you, @Gutza?

**Review by RussKie — APPROVED** [review]

**Comment by Gutza** [comment]
> Does https://ci.appveyor.com/api/buildjobs/whg8axsj2xdhj86v/artifacts/artifacts%2FRelease%2Fpublish%2FGitExtensions-Portable-4.1.0.16396-f9631e23c.zip work for you, @Gutza?

Yes, it works wonderfully, thank you!

**Pull request merged**
